Restore failing intermittently on build

Restore is failing intermittently 

```
F:\workspace\_work\1\s\.dotnet\sdk\5.0.100-preview.5.20251.2\NuGet.targets(128,5): error : Cannot create a file when that file already exists. [F:\workspace\_work\1\s\src\libraries\restore\binplacePackages\binplacePackages.depproj] 
##[error].dotnet\sdk\5.0.100-preview.5.20251.2\NuGet.targets(128,5): error : (NETCORE_ENGINEERING_TELEMETRY=Restore) Cannot create a file when that file already exists. 
```

Looks like a race that is hitting the "Libraries Build Windows_NT allConfigurations x64 Debug" the most often.

Note: if you are looking into this bug make sure to look at all attempts of this leg. About 1/2 of the PRs here retried this failure and it succeeded on retry. Hence looking at the final state of the build will not display this error. 

<!-- runfo report start -->
|Build|Kind|Job Name|
|---|---|---|
|[640103](https://dev.azure.com/dnceng/public/_build/results?buildId=640103)|[PR 36242](https://github.com/dotnet/runtime/pull/36242)|Libraries Build Windows_NT allConfigurations x64 Debug|
|[637444](https://dev.azure.com/dnceng/public/_build/results?buildId=637444)|[PR 36150](https://github.com/dotnet/runtime/pull/36150)|Libraries Build Windows_NT allConfigurations x64 Debug|
|[637444](https://dev.azure.com/dnceng/public/_build/results?buildId=637444)|[PR 36150](https://github.com/dotnet/runtime/pull/36150)|Libraries Build Windows_NT allConfigurations x64 Debug|
|[637444](https://dev.azure.com/dnceng/public/_build/results?buildId=637444)|[PR 36150](https://github.com/dotnet/runtime/pull/36150)|Libraries Build Windows_NT allConfigurations x64 Debug|
|[638863](https://dev.azure.com/dnceng/public/_build/results?buildId=638863)|[PR 36098](https://github.com/dotnet/runtime/pull/36098)|Libraries Build Windows_NT allConfigurations x64 Debug|
|[638863](https://dev.azure.com/dnceng/public/_build/results?buildId=638863)|[PR 36098](https://github.com/dotnet/runtime/pull/36098)|Libraries Build Windows_NT allConfigurations x64 Debug|
|[638849](https://dev.azure.com/dnceng/public/_build/results?buildId=638849)|[PR 36175](https://github.com/dotnet/runtime/pull/36175)|Libraries Build Windows_NT allConfigurations x64 Debug|
|[638786](https://dev.azure.com/dnceng/public/_build/results?buildId=638786)|Rolling|Libraries Build Windows_NT net472 x64 Release|
|[638313](https://dev.azure.com/dnceng/public/_build/results?buildId=638313)|[PR 36098](https://github.com/dotnet/runtime/pull/36098)|Libraries Build Windows_NT allConfigurations x64 Debug|
|[638357](https://dev.azure.com/dnceng/public/_build/results?buildId=638357)|[PR 35366](https://github.com/dotnet/runtime/pull/35366)|Libraries Build Windows_NT allConfigurations x64 Debug|
|[638198](https://dev.azure.com/dnceng/public/_build/results?buildId=638198)|Rolling|Libraries Build Windows_NT allConfigurations x64 Release|

<!-- runfo report end -->

--- a/src/libraries/restore/Directory.Build.props
+++ b/src/libraries/restore/Directory.Build.props
@@ -2,10 +2,13 @@
   <Import Project="..\Directory.Build.props" />
 
   <PropertyGroup>
-      <!-- We need configuration-specific assets files. -->
+    <!-- We need configuration-specific assets files which requires the IntermeidateOutputPath to be set. -->
+    <IntermediateOutputPath>$([MSBuild]::NormalizeDirectory('$(BaseIntermediateOutputPath)', '$(TargetFramework)-$(TargetFrameworkSuffix)-$(Configuration)'))</IntermediateOutputPath>
+    <IntermediateOutputPath Condition="'$(TargetFrameworkSuffix)' == ''">$([MSBuild]::NormalizeDirectory('$(BaseIntermediateOutputPath)', '$(TargetFramework)-$(Configuration)'))</IntermediateOutputPath>
     <RestoreOutputPath>$(IntermediateOutputPath)</RestoreOutputPath>
     <ProjectAssetsFile>$(RestoreOutputPath)/project.assets.json</ProjectAssetsFile>
     <MSBuildProjectExtensionsPath>$(IntermediateOutputPath)</MSBuildProjectExtensionsPath>
+
     <SkipDeriveTargetFrameworks>true</SkipDeriveTargetFrameworks>
     <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
     <AdditionalBuildTargetFrameworks Condition="'$(BuildAllProjects)' == 'true'">netstandard2.0</AdditionalBuildTargetFrameworks>

